File: src/network/lobby/packets/MapPacket.java
package network.lobby.packets;

import misc.compress.CompressBuffer;
import misc.compress.Compressor;
import network.Packet;

public class MapPacket extends Packet
{
	int[][] map;

	public MapPacket(int[][] map)
	{
		this.map = map;
	}

	public int[][] getInts() { return map; }

	public MapPacket(CompressBuffer buffer)
	{
		map = buffer.cutIntIntArray();
	}

	@Override public byte getCID() { return Compressor.MAP_PACKET_CID; }
	@Override public byte[] compress()
	{
		return Compressor.compressIntIntArray(getInts());
	}
}

File: cpp/src/network/packets/MapPacket.cpp
#include "MapPacket.hpp"

#include <misc/Global.hpp>
#include <misc/compress/CompressBuffer.hpp>

MapPacket::MapPacket(const std::vector<std::vector<TileID>>& map)
	: map(map)
{}

MapPacket::MapPacket(CompressBuffer* buffer)
{
	map = buffer->cutMap();
}

std::vector<std::vector<TileID>> MapPacket::getTileIDs() const
{
	return map;
}

CompressID MapPacket::getCompressID() const
{
	return MAP_PACKET_CID;
}

std::string MapPacket::getCompressString() const
{
	return compressMap(getTileIDs());
}
